crates/aranya-client/tests/common/mod.rs

```diff
@@ -156,6 +167,101 @@ impl DevicesCtx {
 
         Ok(team_id)
     }
+
+    fn devices(&self) -> [&DeviceCtx; 5] {
+        [
+            &self.owner,
+            &self.admin,
+            &self.operator,
+            &self.membera,
+            &self.memberb,
+        ]
+    }
+
+    #[instrument(skip(self))]
+    pub async fn add_all_sync_peers(&self, team_id: TeamId) -> Result<()> {
+        let config = SyncPeerConfig::builder().interval(SYNC_INTERVAL).build()?;
+        for device in self.devices() {
+            for peer in self.devices() {
+                if ptr::eq(device, peer) {
+                    continue;
+                }
+                device
+                    .client
+                    .team(team_id)
+                    .add_sync_peer(peer.aranya_local_addr().await?.into(), config.clone())
+                    .await?;
+            }
+        }
+        Ok(())
+    }
+
+    /// NB: This includes the owner role, which is not returned
+    /// by [`Client::setup_default_roles`].
+    #[instrument(skip(self))]
+    pub async fn setup_default_roles(&self, team_id: TeamId) -> Result<DefaultRoles> {
+        self.setup_default_roles_inner(team_id, true).await
+    }
+
+    /// Sets up default roles without creating any management delegations.
+    #[instrument(skip(self))]
+    pub async fn setup_default_roles_without_delegation(
+        &self,
+        team_id: TeamId,
+    ) -> Result<DefaultRoles> {
+        self.setup_default_roles_inner(team_id, false).await
+    }
+
+    #[instrument(skip(self, grant_delegations))]
+    async fn setup_default_roles_inner(
+        &self,
+        team_id: TeamId,
+        grant_delegations: bool,
+    ) -> Result<DefaultRoles> {
+        let owner_role = self
+            .owner
+            .client
+            .team(team_id)
+            .roles()
+            .await?
+            .try_into_owner_role()?;
+        tracing::debug!(owner_role_id = %owner_role.id);
+
+        let setup_roles = self
+            .owner
+            .client
+            .team(team_id)
+            .setup_default_roles(owner_role.id)
+            .await?;
+
+        let roles = setup_roles
+            .into_iter()
+            .chain(iter::once(owner_role))
+            .try_into_default_roles()
+            .context("unable to parse `DefaultRoles`")?;
+        tracing::debug!(?roles, "default roles set up");
+
+        if grant_delegations {
+            let mappings = [
+                // admin -> operator
+                ("admin -> operator", roles.operator().id, roles.admin().id),
```

Comment:
nit: reads better as delegator -> delegate:
```suggestion
                ("admin -> operator", roles.admin().id), roles.operator().id),
```